### ISSUE TITLE
information about rf-detr licene and onnx latency

### ISSUE THREAD

**Issue by SebastianJanampa** [roboflow/rf-detr#865]
Hello Roboflow team,

This question might be a bit odd, but I would like to use the released RF-DETR-XL backbone's weights for academic research, and release (if I successfully get good results) the new models under the Apache License 2.0. 

Also, I was running some inference on my MacBook using RF-Detr Nano with onnx, but its latency is 4X higher than YOLO11 and YOLO26. Is this expected? 

Best,
Sebastian :)

**Comment by Borda** [comment]
Hi Sebastian,

Thanks for reaching out! To answer your questions:

### Model Licensing (RF-DETR-XL):
The RF-DETR-XL and 2XL models are released under the Platform Model License 1.0 (PML-1.0). Because of this, you cannot release derivative models fine-tuned from the XL backbone under the permissive Apache 2.0 license. Any derivatives would still be subject to the PML-1.0 terms. If you want to ensure your academic research can be released under Apache 2.0, we highly recommend using our core model sizes (Nano, Small, Medium, or Large), as those are entirely open-source and licensed under Apache 2.0!

### ONNX Latency on MacBook:
Yes, seeing higher CPU latency compared to YOLO models is expected. While RF-DETR is heavily optimized for real-time GPU inference (e.g., using TensorRT), Transformer-based architectures (like DETR) typically have higher overhead on CPUs (like a MacBook) than purely CNN-based architectures like YOLO11 or YOLO26. Additionally, ONNX Runtime's CPU execution providers tend to be much more highly optimized for standard convolutional layers than for the attention mechanisms used in vision transformers.

Hope this helps with your research, and best of luck with your experiments! Let us know if you have any other questions.

**Comment by SebastianJanampa** [comment]
Hi Borda,
Thank you so much for such a detailed answer. I was checking the RF-DETR configurations, and I noticed that variants from nano to large share the same backbone configuration (num_windows=2). Since RF-DETR models were trained using NAS, do the nano, small, medium, and large variants share the same backbone weights? Also, would it improve performance (at the cost of higher latency) if I set num_windows=1?

**Comment by isaacrob** [comment]
all those sizes were pretrained together so at one point they had the same weights. I don't remember if we finetuned these variants before release; finetuning had limited benefit so I think we may have released seg models without finetuning but the detector models with finetuning. but you can easily check if the weights are the same :) in the pretraining, setting windows=1 would've increased accuracy while increasing latency. it's unclear if finetuning will cause it to forget the ability to use different windows, but it's always something you can try!

**Comment by SebastianJanampa** [comment]
Hi isaacrob,

I checked the weights, and you're right, both detection and segmentation models were finetuned separately.
